roles/apache/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: install apache and necessary mods
+  apt: pkg={{ item }}
+  with_items:
+    - apache2
+    - libapache2-mod-wsgi
+    - libapache2-mod-proxy-html
+
+- name: install ssl cert+key
+  template: src=etc/apache2/{{ item }} dest=/etc/apache2/{{ item }}
+            owner=haproxy group=haproxy mode=0600
+  notify: reload apache
+  with_items:
+    - ssl.key
+    - ssl.crt
+    - ssl.conf
+
+- name: enable apache modules
+  apache2_module: state=present name={{item}}
+  notify:
+    - reload apache
+  with_items:
+    - wsgi
+    - ssl
+    - proxy
+    - proxy_ajp
+    - proxy_http
+    - rewrite
+    - deflate
+    - headers
+    - proxy_balancer
+    - proxy_connect
+    - proxy_html
+
+- name: disable apache status
+  apache2_module: state=absent name=status
+  notify:
+    - reload apache
+
+- name: disable default apache site
+  apache2_site: state=disabled name=000-default
+  notify:
+    - reload apache
+
+- name: check if custom ports.conf
+  exit_status: grep '# This file is managed by Ansible' /etc/apache2/ports.conf
+  run_once: true
+  register: custom_apache_ports_file
+
+- name: create custom ports.conf file
+  template: src=etc/apache2/ports.conf
+            dest=/etc/apache2/ports.conf
+            owner=root
+            group=root
+            mode=0644
+  when: custom_apache_ports_file.status != 0
```

Comment:
Do we expect things external to ansible to be manipulating the ports.conf file? Is that why we can't just template the file and call it good?